pkg/justification/processor.go

```diff
@@ -148,6 +154,21 @@ func (p *Processor) runValidations(req *jvspb.CreateJustificationRequest) error
 		default:
 			err = multierror.Append(err, fmt.Errorf("unexpected justification %v unrecognized", j))
 		}
+
+		if v, ok := p.validators[j.Category]; ok {
+			resp, e := v.Validate(ctx, &jvspb.ValidateJustificationRequest{
+				Justification: j,
+			})
+			if e != nil {
+				err = multierror.Append(err, fmt.Errorf("unexpected error from validator %q: %w", j.Category, e))
+				continue
+			}
+
+			if !resp.Valid {
+				err = multierror.Append(err,
+					fmt.Errorf("failed validation criteria with error %v and warning %v", resp.Error, resp.Warning))
+			}
+		}
```

Comment:
I don't think this is correct, since there's no logic for when the validator doesn't exist for the category. I think you might want:

```suggestion
		v, ok := p.validators[j.Category]
		if !ok {
			err = multierror.Append(err, fmt.Errorf("missing validator for category %q", j.Category))
		}
		
		resp, e := v.Validate(ctx, &jvspb.ValidateJustificationRequest{
			Justification: j,
		})
		if e != nil {
			err = multierror.Append(err, fmt.Errorf("unexpected error from validator %q: %w", j.Category, e))
			continue
		}

		if !resp.Valid {
			err = multierror.Append(err, fmt.Errorf("failed validation criteria with error %v and warning %v", resp.Error, resp.Warning))
		}
```

Reply:
Actually I am a bit confused by the `Category` handling logic in jvs repo, want to confirm the requirements below:

1. In the UI controller, `breakglass` is a category ([code pointer](https://github.com/abcxyz/jvs/blob/a02ba7d38f4b16c203d37e0d879b6c67883ec43a/pkg/controller/controller.go#L170)), but it will be immediately rejected [here](https://github.com/abcxyz/jvs/blob/a02ba7d38f4b16c203d37e0d879b6c67883ec43a/pkg/justification/processor.go#L148)
2. `explanation` is a category allowing free form text

I was thinking the additional plugins will add additional category on top of 1 and 2, and will be displayed as additional drop down items. As a result, `breakglass` and `explanation` would not have any corresponding validator plugins 

Let me know whether my understanding is correct, and also loop in @yolocs to this thread

Reply:
> In the UI controller, breakglass is a category (code pointer), but it will be immediately rejected here

This is actually a bug. I filed an issue https://github.com/abcxyz/jvs/issues/296. I will send a temporary fix for it.

For this PR, we can specially handle the "explanation" category. E.g.

1. Is "explanation"? Yes, generate token based on the request.
2. No, try find a validator. Found? Validate and generate token.
3. Not found. Fail.